plugins/builds/triggers/helpers.js

```diff
@@ -154,35 +150,11 @@ async function createEvent(config) {
         parentEventId,
         groupEventId
     } = config;
-    const { scm } = eventFactory;
-    const payload = {
-        pipelineId,
-        startFrom,
-        type: 'pipeline',
-        causeMessage,
-        parentBuildId
-    };
-
-    if (parentEventId) {
-        payload.parentEventId = parentEventId;
-    }
-
-    // for backward compatibility, this field is optional
```

Comment:
I would like to keep this comment even after refactoring so that we know why this field is optional. Or is it already in a state where we don't need to term backward compatibility?